hal/src/b5som/network/quectel_ncp_client.cpp

```diff
@@ -837,6 +837,18 @@ int QuectelNcpClient::changeBaudRate(unsigned int baud) {
 }
 
 int QuectelNcpClient::initReady() {
+    // Enable flow control and change to runtime baudrate
+    auto runtimeBaudrate = QUECTEL_NCP_DEFAULT_SERIAL_BAUDRATE;
+    uint32_t hwVersion = HW_VERSION_UNDEFINED;
+    auto ret = hal_get_device_hw_version(&hwVersion, nullptr);
+    if (ret == SYSTEM_ERROR_NONE && hwVersion == HAL_VERSION_B5SOM_V003) {
+        CHECK_PARSER(parser_.execCommand("AT+IFC=0,0"));
+    } else {
+        runtimeBaudrate = QUECTEL_NCP_RUNTIME_SERIAL_BAUDRATE;
+        CHECK_PARSER(parser_.execCommand("AT+IFC=2,2"));
```

Comment:
Is it worth checking for an OK here?

Reply:
Yeap, should be `CHECK_PARSER_OK` instead.